blockchain/options.go

```diff
@@ -3,70 +3,94 @@ package blockchain
 import "time"
 
 const (
-	defaultEthEndpoint        = "https://rinkeby.infura.io/00iTrs5PIy0uGODwcsrb"
+	defaultLivechainEndpoint  = "https://rinkeby.infura.io/00iTrs5PIy0uGODwcsrb"
 	defaultSidechainEndpoint  = "https://sidechain-dev.sonm.com"
-	defaultGasPrice           = 20000000000 // 20 Gwei
-	defaultGasPriceSidechain  = 0
+	defaultLivechainGasPrice  = 20000000000 // 20 Gwei
+	defaultSidechainGasPrice  = 0
 	defaultBlockConfirmations = 5
+	defaultLogParsePeriod     = time.Second
 )
 
+// chainOpts describes common options
+// for almost any geth-node connection
+// (live Eth network, rinkeby, SONM sidechain
+// or local geth-node for testing).
+type chainOpts struct {
+	gasPrice           int64
+	endpoint           string
+	logParsePeriod     time.Duration
+	blockConfirmations int64
+}
+
+func (c *chainOpts) newClient() (CustomEthereumClient, error) {
+	return NewClient(c.endpoint)
+}
+
 type options struct {
-	gasPrice             int64
-	gasPriceSidechain    int64
-	apiEndpoint          string
-	apiSidechainEndpoint string
-	logParsePeriod       time.Duration
-	blockConfirmations   int64
+	livechain *chainOpts
```

Comment:
`livechain` is abuse naming.
we have agreement to name this `masterchain`.

Reply:
I'll do it with upcoming PR to not make out-of-scope changes here.